SixTrack/sixtrack.s

```diff
@@ -67069,6 +67091,51 @@ c$$$     &           myalphay * cos(phiy))
 !
 !cccccccccccccccccccccccccccccccccc
 
+! Function to calculate the electron density in a material
+! Should give the number per cubic meter
+      function CalcElectronDensity(AtomicNumber, Density, AtomicMass)
+      implicit none
+      double precision AtomicNumber, Density, AtomicMass
+      double precision Avogadro
+      double precision CalcElectronDensity
+      double precision PartA, PartB
+      Avogadro = 6.022140857e23
+      PartA = AtomicNumber * Avogadro * Density
+      !1e-6 factor converts to n/m^-3
+      PartB = AtomicMass * 1e-6
+      CalcElectronDensity = PartA/PartB
+      return
+      end
+
+! Function to calculate the plasma energy in a material
+! CalculatePlasmaEnergy = (PlanckConstantBar * sqrt((ElectronDensity *(ElectronCharge**2)) / (ElectronMass * FreeSpacePermittivity)))/ElectronCharge*eV;
+      function CalcPlasmaEnergy(ElectronDensity)
+      implicit none
+      double precision ElectronDensity
```

Comment:
Thank you for the implicit none. Note that it may be possible to make the constants actual constants (parameters)?